Add async overload for search API

Sherlouk opened the pull request:
# Pull Request

⚠️ I've marked this as draft as it current works to provide an example of how we could add async/await support to the library. I am more than happy to expand this to **all** currently supported APIs.

## Related issue

Relates to #332 (doesn't close as only adds support for one API)

## What does this PR do?
- Adds an async/await override to an existing search API providing users with more choice.

## PR checklist
Please check if your PR fulfills the following requirements:
- [x] Does this PR fix an existing issue, or have you listed the changes applied in the PR description (and why they are needed)?
- [x] Have you read the contributing guidelines?
- [x] Have you made sure that the title is accurate and descriptive of the changes?

Thank you so much for contributing to Meilisearch!

aronbudinszky commented on Sources/MeiliSearch/Indexes.swift:
```diff
@@ -26,7 +26,7 @@ public struct Indexes {
 
   // Search methods
-  private let search: Search
+  fileprivate let search: Search
```
This change is actually not required - `private` will work just as fine.

Sherlouk replied:
Let me double check this, I thought I had a compiler error when I tested (due to use of an extension) but will double check and revert if possible.

In respect of "I have not actually tested it though" - I'm using this in a production application and is working fine for me 🙂 Though I'll keep an eye on it!

aronbudinszky replied:
Yeah that's fair, agreed. (If there is value in hiding these properties internally we could still use protocols for that - though not sure if it is necessary.)

Only final point that I'd make then is that I would not explicitly state `internal` - given that this is the default anyway and not used on other internal properties.

Sherlouk replied:
Fair point, that's a personal preference but deviates from standard practice for the project. I'll amend in a follow up PR

aronbudinszky reviewed (approved):
Aside from the minor comment regarding `fileprivate` all looks good to me!

I have not actually tested it though, but assume it does work given that the unit test passes.

curquiza commented:
Thank you @aronbudinszky for reviewing and @Sherlouk for the PR

Can you confirm there is no breaking for the user? 😊 
Is this change worth adding some documentation to the README? Or is ti really "usage transparent"?

Sherlouk commented:
This is not a breaking change, no.
I do think it's worth updating documentation with a quick example, but it is also obvious when using.

Sherlouk commented:
I've updated documentation, tweaked implementation, and opened up the PR. This does only add support for one API (search), but if we're happy then I can either:

1. Merge this PR, and then open a new one with support for all other APIs
2. Keep this PR open, until I've added support for all other APIs
3. Merge this PR, but open support for each other API one PR at a time (as I work on them with other features).

Clem, thoughts?

curquiza commented:
Let's go with #1, not with #3 indeed 😄

curquiza reviewed (approved):
bors merge

aronbudinszky commented:
Btw retroactively confirming there is no breaking change. :)
